### PR TITLE
Add conversions to/from std wire types

### LINKED ISSUE #147: Use standard library types where possible
I've noticed smoltcp defines its own `Ipv4Address` and `IpEndpoint` types instead of using the standard `Ipv4Addr` and `SocketAddr` types. Is there any good reason for this? It's kind of annoying to have to convert between the two.

### PR DESCRIPTION
**Pull request by canndrew**
Fixes #147

### CODE REVIEW

**Comment by whitequark** [comment]
This broke CI.

**Comment by canndrew** [comment]
CI passes now.

File: src/wire/ip.rs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 use core::fmt;
 use core::convert::From;
+#[cfg(all(feature = "std", feature = "proto-ipv4", feature = "proto-ipv6"))]
```
Comment (whitequark) [inline src/wire/ip.rs]: I'd prefer if we had no conditional `use` statements.

Comment (canndrew) [reply src/wire/ip.rs]: I've changed it to use global paths instead. eg. `::std::net::Ipv4Addr`.

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**